modules/core/core-scheduler/src/test/java/com/enonic/xp/impl/scheduler/distributed/RescheduleTaskTest.java

```diff
@@ -92,38 +113,199 @@ void tearDown()
+
+        when( taskService.submitTask( isA( SubmitTaskParams.class ) ) ).thenThrow( new RuntimeException() );
+
+        for ( int i = 0; i <= 10; i++ )
+        {
+            createAndRunTask();
+        }
+        verify( taskService, times( 11 ) ).submitTask( taskCaptor.capture() );
+    }
+
+    @Test
+    public void submitJobAsUser()
+    {
+        final Instant now = Instant.now();
+        final PrincipalKey user = PrincipalKey.ofUser( IdProviderKey.createDefault(), "my-user" );
+
+        ScheduledJob job1 = mockOneTimeJob( "job1", now.minus( 1, ChronoUnit.SECONDS ), user );
+
+        when( schedulerService.list() ).thenReturn( List.of( job1 ) );
+
+        final Node node = mockNode();
+        when( nodeService.update( isA( UpdateNodeParams.class ) ) ).thenReturn( node );
+
+        when( taskService.submitTask( isA( SubmitTaskParams.class ) ) ).thenReturn( TaskId.from( "1" ) );
+        when( securityService.authenticate( tokenCaptor.capture() ) ).thenReturn( mock( AuthenticationInfo.class ) );
 
         createAndRunTask();
 
-        verify( schedulerExecutorService, times( 2 ) ).schedule( taskCaptor.capture(), anyLong(), isA( TimeUnit.class ) );
+        assertEquals( "default", tokenCaptor.getValue().getIdProvider().toString() );
+        verify( taskService, times( 1 ) ).submitTask( taskCaptor.capture() );
+    }
+
+    @Test
+    public void submitCronJob()
+    {
+        ScheduledJob job1 = mockCronJob( "job1", "* * * * *", Instant.parse( "2021-02-26T10:44:33.170079900Z" ) );
+        ScheduledJob job2 = mockCronJob( "job2", "* * * * *", Instant.now() );
+
+        when( schedulerService.list() ).thenReturn( List.of( job1, job2 ) );
+
+        final Node node = mockNode();
+        when( nodeService.update( isA( UpdateNodeParams.class ) ) ).thenReturn( node );
 
-        assertEquals( 2, taskCaptor.getAllValues().size() );
-        assertEquals( "task2", taskCaptor.getAllValues().get( 0 ).getName() );
-        assertEquals( "task3", taskCaptor.getAllValues().get( 1 ).getName() );
+        when( taskService.submitTask( isA( SubmitTaskParams.class ) ) ).thenReturn( TaskId.from( "1" ) ).thenReturn( TaskId.from( "2" ) );
+        when( securityService.authenticate( tokenCaptor.capture() ) ).thenReturn( mock( AuthenticationInfo.class ) );
+
+        createAndRunTask();
+
+        verify( taskService, times( 1 ) ).submitTask( taskCaptor.capture() );
+        assertEquals( "job1", taskCaptor.getValue().getDescriptorKey().getName() );
+    }
+
+    @Test
+    public void jobWasRemoved()
+        throws InterruptedException
+    {
+        final Instant plus = null;
+
+        ScheduledJob job1 = mockCronJob( "job1", "* * * * *", plus );
+        ScheduledJob job2 = mockCronJob( "job2", "* * * * *", plus );
+
+        when( schedulerService.list() ).thenReturn( List.of( job1, job2 ) );
+
+        createAndRunTask();
+
+        Thread.sleep( 61000 );
```

Comment:
Actually SheduleJob accepts SheduleCalenar, that is interface and it is possible to mock CronCalendar, too. No need to wait 60 seconds